java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -396,13 +396,112 @@ public void positiveCaseWithFlagNameAsVariable() throws IOException {
     }
   }
 
+  @Test
+  public void caseWithFlagNameAsVariableWithReturnTypeRegex() throws IOException {
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "src/test/resources/config/properties_test_return.json");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+          BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.allowBreakingChanges();
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+      bcr = addHelperClasses(bcr);
+      bcr.addInputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              "private static final String STALE_FLAG_CONSTANTS = \"STALE_FLAG\";",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  if (experimentation.isToggleEnabled(STALE_FLAG_CONSTANTS)) { int a = 1; }",
+              "     else { int b = 2;}",
+              "  if (experimentation.isToggleDisabled(STALE_FLAG_CONSTANTS)) { return \"X\"; }",
+              "     else { return \"Y\";}",
+              " }",
+              "}")
+          .addOutputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  if (experimentation.isToggleEnabled(STALE_FLAG_CONSTANTS)) { int a = 1; }",
```

Comment:
Probably worth a comment as to why this isn't simplified. I get that the `returnType` is not marked as `bool.*` (also: any reason for that vs `boolean`?), but I am not sure I'll remember that when looking at this test case in isolation years down the line ;)

Reply:
+1. useful to have additional comments for all the tests that are added here.

Reply:
Done! 

>  I get that the returnType is not marked as bool.* (also: any reason for that vs boolean?)

Since I've used regex matching for both return and receiver type, just thought a test making it explicit is worth having.